Maple2.Database/Storage/Game/GameStorage.Club.cs

```diff
@@ -1,64 +1,148 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using Maple2.Database.Extensions;
+﻿using Maple2.Database.Extensions;
+using Maple2.Model.Enum;
 using Maple2.Model.Game;
-using Club = Maple2.Model.Game.Club;
-using ClubMember = Maple2.Model.Game.ClubMember;
+using Maple2.Model.Game.Party;
+using Maple2.Tools.Extensions;
+using Z.EntityFramework.Plus;
+using Club = Maple2.Model.Game.Club.Club;
+using ClubMember = Maple2.Model.Game.Club.ClubMember;
 
 namespace Maple2.Database.Storage;
 
 public partial class GameStorage {
     public partial class Request {
         public Club? GetClub(long clubId) {
-            Model.Club? model = Context.Club.Find(clubId);
-            if (model == null) {
+            return Context.Club.Find(clubId);
+        }
+
+        public bool ClubExists(string clubName = "") {
+            return Context.Club.Any(club => club.Name == clubName);
+        }
+
+        public bool ClubExists(long clubId = 0) {
+            return Context.Club.Any(club => club.Id == clubId);
+        }
+
+        public IList<long> ListClubs(long characterId) {
+            return Context.ClubMember.Where(member => member.CharacterId == characterId)
+                .Select(member => member.ClubId)
+                .ToList();
+        }
+
+        public Club? CreateClub(string name, long leaderId, List<PartyMember> partyMembers) {
```

Comment:
Let's simplify this to `List<PlayerInfo>` instead of `List<PartyMember>`. I would actually prefer `List<long> characterIds` if that works.